assets/js/editor-components/color-panel/index.tsx

```diff
@@ -0,0 +1,190 @@
+/* eslint-disable @typescript-eslint/ban-ts-comment */
+/**
+ * External dependencies
+ */
+import { useMemo } from '@wordpress/element';
+import {
+	store as blockEditorStore,
+	getColorClassName,
+	InspectorControls,
+	useBlockEditContext,
+	__experimentalColorGradientSettingsDropdown as ColorGradientSettingsDropdown,
+	__experimentalUseMultipleOriginColorsAndGradients as useMultipleOriginColorsAndGradients,
+} from '@wordpress/block-editor';
+import { useSelect, useDispatch } from '@wordpress/data';
+
+interface ColorSetting {
+	colorValue: string | undefined;
+	onColorChange: ( value: string ) => void;
+	label: string;
+	resetAllFilter: () => void;
+}
+
+interface ColorAttributes {
+	[ key: string ]: {
+		[ key: string ]: string;
+	};
+}
+
+export interface CustomColorsMap {
+	[ key: string ]: {
+		label: string;
+		context: string;
+	};
+}
+
+export interface ColorPaletteOption {
+	name: string;
+	slug: string;
+	color: string;
+}
+
+export interface GradientPaletteOption {
+	name: string;
+	gradient: string;
+	slug: string;
+}
+
+interface ColorGradientOptionsColorItem {
+	name: string;
+	colors: ColorPaletteOption[];
+}
+
+interface ColorGradientOptionsGradientItem {
+	name: string;
+	gradients: GradientPaletteOption[];
+}
+
+interface ColorGradientOptionsItems {
+	colors: [ ColorGradientOptionsColorItem ];
+	gradients: [ ColorGradientOptionsGradientItem ];
+}
```

Comment:
What do you think about moving all these types, or at least the ones which are exported, to a separate `types.ts` file? I feel it helps keep the code more organized and helps with discoverability.